csharp/src/Apache.Arrow/Memory/LongMemory.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Apache.Arrow.Memory
+{
+    internal readonly struct LongMemory<T>
+    {
+        private readonly T[] _array;
```

Comment:
This type probably needs to hold a raw pointer rather than an array, arrays in .NET are limited to 2 GiB, which is why the new long types are needed. It might be worth reaching out to Curt who's the Arrow maintainer to say you're looking into this and asking if it's possible for him to share the work he started on.

Reply:
(https://github.com/apache/arrow/issues/38086#issuecomment-2275679374)

Reply:
Actually, thinking about this some more, rather than a raw pointer it probably needs to hold a LongMemoryManager.